Library/ViewModels/PledgeViewModel.swift

```diff
@@ -498,15 +498,25 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       scaFlowCompletedWithSuccess.combineLatest(with: creatingContext).ignoreValues()
     )
 
+    let checkoutId = createBackingEvents.values()
+      .map(first)
+      .map(decompose)
+
     let thanksPageData = createBackingDataAndIsApplePay
-      .map { data, isApplePay -> ThanksPageData in
-        let checkoutPropsData = checkoutPropertiesData(from: data, isApplePay: isApplePay)
+      .takeWhen(createBackingCompletionEvents)
+      .combineLatest(with: checkoutId)
```

Comment:
So this change is actually the same thing as before, although we've moved things around this still combines the latest emissions from the two signals. Basically it reads like this "emit `createBackingDataAndIsApplePay` when `createBackingCompletionEvents` emits and combine its emission with `checkoutId`" which means that whenever `checkoutId` or `createBackingCompletionEvents` emit then this will emit. We just want to be explicit that this should only emit _at all_ when `createBackingCompletionEvents` emits. 

This is actually achievable just by moving things around slightly and carrying the `checkoutId` all the way through to `createBackingCompletionEvents` which will require a few changes higher up in the chain:

```swift
let thanksPageData = createBackingDataAndIsApplePay
  .takePairWhen(createBackingCompletionEvents)
  .map(unpack)
  .map { data, isApplePay, checkoutId -> ThanksPageData in
    let checkoutPropsData = checkoutPropertiesData(
      from: data,
      checkoutId: checkoutId,
      isApplePay: isApplePay
    )

    return (data.project, data.reward, checkoutPropsData)
  }
```